src/server/api/routers/club/members.ts

```diff
@@ -16,9 +16,18 @@ export const memberRouter = createTRPCRouter({
       const club = await ctx.db
         .selectFrom("Club")
         .selectAll()
-        .where("joinCode", "==", joinCode)
+        .where("joinCode", "=", joinCode)
         .executeTakeFirstOrThrow();
 
+      const membership = await ctx.db
+        .selectFrom("ClubMembership")
+        .where("ClubMembership.clubID", "=", club.ID)
+        .where("ClubMembership.userID", "=", userID)
+        .select("ClubMembership.clubID")
+        .executeTakeFirst();
+
+      if (membership?.clubID) return club;
```

Comment:
Might be worth throwing an error if a user tries to join a a club they are not a member of?